calico_cni_test.go

```diff
@@ -190,7 +190,7 @@ var _ = Describe("CalicoCni", func() {
 					if err := testutils.CreateHostVeth(containerID, "", "", hostname); err != nil {
 						panic(err)
 					}
-					_, session, _, _, _, contNs, err := testutils.CreateContainerWithId(netconf, "", testutils.TEST_DEFAULT_NS, "", containerID)
+					_, session, _, _, _, contNs, err := testutils.CreateContainer(netconf, "", testutils.TEST_DEFAULT_NS, "")
```

Comment:
This is where the beauty of this test code truly shines. 

`CreateContainerWithId` just called `CreateContainerNamespaceWithCid` with that ID, and all the latter did was return it: https://github.com/projectcalico/cni-plugin/blob/fb6ab1f8750a380ffd49734c5c2fc0ac08de284b/testutils/utils.go#L237-L239

The Container ID doesn't _have_ to match the veth. `CreateContainer` is returning it as a sample value you can use when you call `RunCNIPluginWithId`, since it what we know Kubernetes will do. But other orchestrators may use other values. So instead of passing a containerID to CreateContainer that will just get passed back to you, I've switched it so that it gives you a containerID back that *you don't have to use if you don't want to*.

I've updated the function descriptions to better explain this. And am considering if we don't even want to return any ContainerID at all and have the caller use some helper if it wants to get the ContainerID that kubernetes would usually assign.